packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +88,117 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>
+                    <div className="options">
+                        <input
+                            type="text"
+                            placeholder="Enter a custom font here"
+                            id="customFont"
+                            onKeyDown={(event) => {
+                                event.key === "Enter" && this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },
+                                          Number(this.state.value)
+                                      ),
+                                      this.updateCustomValue(),
+                                      this.forceUpdate())
+                                    : null;
+                            }}
+                            onBlur={() => {
+                                this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },
+                                          Number(this.state.value)
+                                      ),
+                                      this.updateCustomValue(),
+                                      this.forceUpdate())
+                                    : null;
+                            }}
+                        />
+                    </div>
                 </div>
-                <div className="options">
-                    <select onChange={(evt) => this.updateValue(evt.target.value)} value={this.state.value ?? ""}>
-                        {this.props.options.map((option, i) => {
-                            return (
-                                <option selected={option.selected} key={option.label + i} value={option.value} title={option.label}>
-                                    {option.label}
-                                </option>
-                            );
-                        })}
-                    </select>
+            );
+        } else {
+            return (
+                <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>

```

Comment:
Those lines are repeated in both if and else. Can you extract them out of those loops?

Reply:
What would the syntax for that be? If I extract that part it's missing a closing div tag, but if I add that in then I don't think I'd be able to use that within the if else statements right? Since technically there is no closing div tag for the first div because there's more code added within it?